cmake/emu/native.cmake

```
@@ -3,7 +3,7 @@
 add_custom_target(run_native
   COMMAND
   ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_EXE_NAME}
-  DEPENDS ${logical_target_for_zephyr_elf}
+  DEPENDS native_executable
```

```suggestion
  DEPENDS native_executable $<$<TARGET_EXISTS:native_runner_executable>:native_runner_executable> 
```
will establish the missing dependency when building Zephyr as a native library.